File: LC_1431.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        
        int max = Integer.MIN_VALUE;
        List<Boolean> canBeGreatest = new ArrayList<>();
        
        for (int candy : candies)
            if (candy > max)
                max = candy;
        
        for (int i = 0; i < candies.length; i++)
            if (candies[i] + extraCandies >= max)
                canBeGreatest.add(true);
            else
                canBeGreatest.add(false);
        
        return canBeGreatest;
    }
}

File: LC_1431.py
# 1431. Kids With the Greatest Number of Candies

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        result = [False] * len(candies)
        initial_max = max(candies)
        
        for i in range(len(candies)):
            result[i] = candies[i] + extraCandies >= initial_max
        
        return result
        